test/frontend/tests/workflow-tests.js

```diff
@@ -18,87 +20,61 @@ navigator.serviceWorker.register('/serviceWorker.js');
 
 describe('Upload / Download flow', function() {
   this.timeout(0);
-  it('can only download once by default', async function() {
+  it('can download', async function() {
     const fs = new FileSender();
-    const file = await fs.upload(archive);
-    const fr = new FileReceiver({
-      secretKey: file.toJSON().secretKey,
-      id: file.id,
-      nonce: file.keychain.nonce,
-      requiresPassword: false
-    });
+    const file = await fs.upload(archive, correctToken);
+    const fr = new FileReceiver(
+      {
+        secretKey: file.toJSON().secretKey,
+        id: file.id,
+        nonce: file.keychain.nonce,
+        requiresPassword: false
+      },
+      correctToken
+    );
     await fr.getMetadata();
     await fr.download(options);
-
-    try {
-      await fr.download(options);
-      assert.fail('downloaded again');
-    } catch (e) {
-      assert.equal(e.message, '404');
-    }
   });
 
   it('downloads with the correct password', async function() {
     const fs = new FileSender();
-    const file = await fs.upload(archive);
+    const file = await fs.upload(archive, correctToken);
     await file.setPassword('magic');
-    const fr = new FileReceiver({
-      secretKey: file.toJSON().secretKey,
-      id: file.id,
-      url: file.url,
-      nonce: file.keychain.nonce,
-      requiresPassword: true,
-      password: 'magic'
-    });
+    const fr = new FileReceiver(
+      {
+        secretKey: file.toJSON().secretKey,
+        id: file.id,
+        url: file.url,
+        nonce: file.keychain.nonce,
+        requiresPassword: true,
+        password: 'magic'
+      },
+      correctToken
+    );
     await fr.getMetadata();
     await fr.download(options);
     assert.equal(fr.state, 'complete');
   });
 
-  it('blocks invalid passwords from downloading', async function() {
```

Comment:
Can we rework this with the invalid token?

Reply:
The problem here is that either the bearer token or the password are passed in the same way: as the `Authenticaion` header. 

In the previous version the `hmac` auth method was used, which required the password. To implement it, we have to pass the password in a different way (probably, as a different header?). 

Should I implement it?

Reply:
I thought we can add test case for the token with invalid issuer or expiration date, so the test will throw an expected exception